### PR TITLE
[12.0] add crm team parent field

### LINKED ISSUE #241: Adding a Parent-Child hierarchy on Sales Teams
Hello,

I'm actually missing a notion to indicate that a Sales Team is part of another Sales Team (ex: Sales Team Berlin is part of Sales Team Germany).
Therefore, we would like to add a 'Parent' field on the Sales Team form. It can afterwards be used in the reports to look at the pipeline / sales at parent level...

@qgroulard will make a pull request for this.

### PR DESCRIPTION
**Pull request by qgroulard**
Add a 'Parent Team' field on the Sales Team form.

fix https://github.com/OCA/crm/issues/241

### CODE REVIEW

File: crm_team_parent/models/crm_team.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+class CrmTeam(models.Model):
+
+    _inherit = "crm.team"
+    _parent_store = True
+    parent_path = fields.Char(index=True)
+    parent_id = fields.Many2one(comodel_name="crm.team", string="Parent Team")
+    child_ids = fields.One2many(
+        comodel_name="crm.team", inverse_name="parent_id", string="Children Sales Team"
```
Comment (sbidoul) [inline crm_team_parent/models/crm_team.py]:
```suggestion
        comodel_name="crm.team", inverse_name="parent_id", string="Children teams"
```

File: crm_team_parent/models/crm_team.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class ParentLoopError(ValidationError):
+    pass
+
+
+class CrmTeam(models.Model):
+
+    _inherit = "crm.team"
+    _parent_store = True
+    parent_path = fields.Char(index=True)
+    parent_id = fields.Many2one(comodel_name="crm.team", string="Parent Team")
```
Comment (sbidoul) [inline crm_team_parent/models/crm_team.py]:
```suggestion
    parent_id = fields.Many2one(comodel_name="crm.team", string="Parent team")
```

File: crm_team_parent/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Crm Team Parent",
+    "summary": """
+        Add a 'Parent' field on the Sales Team form.""",
```
Comment (sbidoul) [inline crm_team_parent/__manifest__.py]:
```suggestion
        Add a parent field on sales teams.""",
```

**Review by sbidoul — APPROVED** [review]
LGTM. A few minor comments.

**Review by gva-acsone — APPROVED** [review]
👍 Functional test

**Comment by sbidoul** [comment]
@qgroulard can you rebase and squash in one commit saying `[ADD] crm_team_parent`

**Comment by qgroulard** [comment]
@sbidoul  30c93a628eb5573a49c7251b9602e964438a2460

**Review by ThomasBinsfeld — APPROVED** [review]

**Comment by sbidoul** [comment]
@OCA/crm-sales-marketing-maintainers is this good for merging?

File: crm_team_parent/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Crm Team Parent",
+    "summary": """
+        Add a parent field on sales teams.""",
+    "version": "12.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "ACSONE SA/NV,Odoo Community Association (OCA)",
+    "website": "https://github.com/acsone/crm",
```
Comment (rafaelbn) [inline crm_team_parent/__manifest__.py]: Please fix URL : https://github.com/OCA/crm

**Review by rafaelbn — COMMENTED** [review]
Hi! Can I ask for the real use case of this module? Just curiosity.

**Comment by qgroulard** [comment]
The use case is a company with a strong hierarchy regarding their sales teams.
They have several teams with several sub-teams.
A member of a team has access to  the sub-teams resources.

**Review by rafaelbn — APPROVED** [review]

**Comment by rafaelbn** [comment]
This is not a FIX is a new module. I guees /ocabot merge

Take care about hieracy and remember that Odoo don't allow to have 1 sales person in diferent teams

**Comment by sbidoul** [comment]
The test failure is unrelated: the 12.0 branch is red since December last year.

**Comment by sbejaoui** [comment]
Let see if a rebase fix the unrelated failing test issue

**Pull request merged**